File: app/src/main/java/com/my/televip/virtuals/messenger/FileLoadOperation.java
package com.my.televip.virtuals.messenger;

import com.my.televip.obfuscate.ArgsResolver;
import com.my.televip.obfuscate.Obfuscate;

import de.robv.android.xposed.XposedHelpers;

public class FileLoadOperation {

    private final Object fileOperation;

    public FileLoadOperation(Object fileOperation){ this.fileOperation = fileOperation; }

    public void setDownloadChunkSizeBig(int v){
        XposedHelpers.setIntField(fileOperation, Obfuscate.getFieldName("FileLoadOperation", "downloadChunkSizeBig"), v);
    }

    public void setMaxDownloadRequests(int v){
        XposedHelpers.setIntField(fileOperation, Obfuscate.getFieldName("FileLoadOperation", "maxDownloadRequests"), v);
    }

    public void setMaxDownloadRequestsBig(int v){
        XposedHelpers.setIntField(fileOperation, Obfuscate.getFieldName("FileLoadOperation", "maxDownloadRequestsBig"), v);
    }

    public void setMaxCdnParts(int v){
        XposedHelpers.setIntField(fileOperation, Obfuscate.getFieldName("FileLoadOperation", "maxCdnParts"), v);
    }

}

File: app/src/main/java/com/my/televip/virtuals/messenger/FileLoadOperation.kt
package com.my.televip.virtuals.messenger

import com.my.televip.obfuscate.Obfuscate
import de.robv.android.xposed.XposedHelpers

class FileLoadOperation(private val fileOperation: Any?) {

    fun setDownloadChunkSizeBig(v: Int) {
        val target = fileOperation ?: return
        XposedHelpers.setIntField(target, Obfuscate.getFieldName("FileLoadOperation", "downloadChunkSizeBig"), v)
    }

    fun setMaxDownloadRequests(v: Int) {
        val target = fileOperation ?: return
        XposedHelpers.setIntField(target, Obfuscate.getFieldName("FileLoadOperation", "maxDownloadRequests"), v)
    }

    fun setMaxDownloadRequestsBig(v: Int) {
        val target = fileOperation ?: return
        XposedHelpers.setIntField(target, Obfuscate.getFieldName("FileLoadOperation", "maxDownloadRequestsBig"), v)
    }

    fun setMaxCdnParts(v: Int) {
        val target = fileOperation ?: return
        XposedHelpers.setIntField(target, Obfuscate.getFieldName("FileLoadOperation", "maxCdnParts"), v)
    }
}
